lib/install/validate/update.go

```diff
@@ -39,12 +47,58 @@ func (v *Validator) MigrateConfig(ctx context.Context, conf *config.VirtualConta
 }
 
 func (v *Validator) migrateData(ctx context.Context, conf *config.VirtualContainerHostConfigSpec) (*config.VirtualContainerHostConfigSpec, error) {
+	bn, err := buildNumber(conf.Version)
+	if err != nil {
+		v.NoteIssue(err)
+		return nil, err
+	}
+
+	// migrate network
+	// change "external" to "public"
+	if n := conf.Networks["external"]; n != nil {
+		n.Name = "public"
+		n.Network.Name = "public"
+		conf.Networks["public"] = n
+		delete(conf.Networks, "external")
+	}
+
+	if bn == BuildNumber070 {
+		// migrate target
+		conf.Target = conf.Target_.String()
+		// copy over gateway and nameservers if the endpoint
+		// was not dynamic
+		for _, n := range conf.Networks {
+			if !n.Static && (n.IP == nil || ip.IsUnspecifiedIP(n.IP.IP)) {
+				continue
+			}
+			n.Network.Gateway = n.Network.Gateway_
+			n.Network.Nameservers = n.Network.Nameservers_
+		}
+	}
```

Comment:
so these two blocks are all migrate data from .7 to .8. We should have a method explicitly named with that, and called in this migratedata method, that will help us while we drop support for .7